### PR TITLE
Team box and leaderboard folder typescript

### PR DESCRIPTION
**Pull request by Joshowens16**
leaderboard, team one and team two boxes typescriptified

### CODE REVIEW

File: src/components/Leaderboard/leaderboard.types.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface Player {
+    id: string;
+    username: string;
```
Comment (ocjarman) [inline src/components/Leaderboard/leaderboard.types.ts:3]: so clean! will start using same convention

File: src/components/teamBoxes/TeamOneBox.tsx [hunk 2]
```diff
@@ -5,50 +5,53 @@ import { get, ref, set, child, onValue, onDisconnect } from 'firebase/database';
...
-        //If players already exist as team one spymasters:
+        // If players already exist as team one spymasters:
         if (snapshot.exists()) {
-          //'teamOneSpymasters' sets the spymasers id's to an array
-          const teamOneSpymaster = snapshot.val();
-          //Now we can check if the player is a spymaster, if they are, for now we just console log
-          if (teamOneSpymaster.playerId === playerId) {
+          // 'teamOneSpymasters' sets the spymasers id's to an array
+          const teamOneSpymasterGet = snapshot.val();
+          // Now we can check if the player is a spymaster, if they are, for now we just console log
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamOneBox.tsx:51]: should we start cleaning up some of these comments? i think they're from pseudocode prior to writing this logic

File: src/components/teamBoxes/TeamOneBox.tsx [hunk 3]
```diff
@@ -67,34 +70,36 @@ const TeamOneBox = () => {
...
     if (teamTwoOpsAndSpys && teamTwoOpsAndSpys.operatives) {
       teamTwoOperatives = Object.keys(teamTwoOpsAndSpys.operatives);
     }
-    //If a player is on team 2, they cannot join this team
+    // If a player is on team 2, they cannot join this team
     if (
       (teamTwoSpymaster && teamTwoSpymaster.playerId === playerId) ||
       (teamTwoOperatives && teamTwoOperatives.includes(playerId))
     ) {
+      // tslint:disable-next-line:no-console
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamOneBox.tsx:91]: not sure - is this doing something here?

File: src/components/teamBoxes/TeamOneBox.tsx [hunk 4]
```diff
@@ -67,34 +70,36 @@ const TeamOneBox = () => {
...
+        // If players already exist as team one operatives:
         if (snapshot.exists()) {
-          //Now we can check if the player is an operative, if they are for now we just console log
-          const teamOneOperatives = Object.keys(snapshot.val());
-          if (teamOneOperatives.includes(playerId)) {
+          // Now we can check if the player is an operative, if they are for now we just console log
+          const teamOneOperativesGet = Object.keys(snapshot.val());
+          if (teamOneOperativesGet.includes(playerId)) {
             // later we should probably refactor thisso that something on the UI is triggered
+            // tslint:disable-next-line:no-console
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamOneBox.tsx:102]: same question as line 91

File: src/components/teamBoxes/TeamTwoBox.tsx [hunk 5]
```diff
@@ -5,50 +5,53 @@ import { child, get, onDisconnect, onValue, ref, set } from 'firebase/database';
...
+      // Here we want to check if a player is already a spymaster, so that they cannot join both
       await get(teamTwoSpymasterRef).then((snapshot) => {
-        //If players already exist as team one spymasters:
+        // If players already exist as team one spymasters:
         if (snapshot.exists()) {
-          //'teamTwoSpymasters' sets the spymasers id's to an array
-          const teamTwoSpymaster = snapshot.val();
-          //Now we can check if the player is a spymaster, if they are, for now we just console log
-          if (teamTwoSpymaster.playerId === playerId) {
+          // 'teamTwoSpymasters' sets the spymasers id's to an array
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamTwoBox.tsx:49]: i forgot to take this comment out on my last PR - this 'array' comment is no longer accurate!

File: src/components/teamBoxes/TeamTwoBox.tsx [hunk 6]
```diff
@@ -5,50 +5,53 @@ import { child, get, onDisconnect, onValue, ref, set } from 'firebase/database';
...
+          const teamTwoSpymasterSnap = snapshot.val();
+          // Now we can check if the player is a spymaster, if they are, for now we just console log
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamTwoBox.tsx:51]: comment inaccurate

File: src/components/teamBoxes/TeamTwoBox.tsx [hunk 7]
```diff
@@ -68,35 +71,38 @@ const TeamTwoBox = () => {
...
+      // tslint:disable-next-line:no-console
       console.log('Cannot join the other team!');
     } else {
-      //Here we want to check if a player is already an operative, so that they cannot join both.
+      // Here we want to check if a player is already an operative, so that they cannot join both.
       await get(teamTwoOperativesRef).then((snapshot) => {
-        //If players already exist as team one operatives:
+        // If players already exist as team one operatives:
         if (snapshot.exists()) {
-          //Now we can check if the player is an operative, if they are for now we just console log
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamTwoBox.tsx:96]: same comment as above

File: src/components/teamBoxes/TeamTwoBox.tsx [hunk 8]
```diff
@@ -68,35 +71,38 @@ const TeamTwoBox = () => {
...
-          const teamTwoOperatives = Object.keys(snapshot.val());
-          if (teamTwoOperatives.includes(playerId)) {
+          // Now we can check if the player is an operative, if they are for now we just console log
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamTwoBox.tsx:101]: same comment as above

File: src/components/teamBoxes/TeamTwoBox.tsx [hunk 9]
```diff
@@ -68,35 +71,38 @@ const TeamTwoBox = () => {
...
+          const teamTwoOperativesSnap = Object.keys(snapshot.val());
+          if (teamTwoOperativesSnap.includes(playerId)) {
             // later we should probably refactor thisso that something on the UI is triggered
+            // tslint:disable-next-line:no-console
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamTwoBox.tsx:105]: needed?

**Review by ocjarman — CHANGES_REQUESTED** [review]
looks great! 

there are some comments throughout these files that i haven't touched in previous PR's, but now that i think the team boxes logic is pretty set, we might want to clean them up while we're here :) what do you think?

File: src/components/teamBoxes/TeamOneBox.tsx (hunk 2 above)
Comment (Joshowens16) [reply src/components/teamBoxes/TeamOneBox.tsx:51]: Sounds good, I'll start takin some out!

File: src/components/teamBoxes/TeamOneBox.tsx (hunk 3 above)
Comment (Joshowens16) [reply src/components/teamBoxes/TeamOneBox.tsx:91]: For some reason typescript gives me a warning on console logs so this is just me disabling the rule but I can remove them!

File: src/components/teamBoxes/TeamTwoBox.tsx (hunk 6 above)
Comment (Joshowens16) [reply src/components/teamBoxes/TeamTwoBox.tsx:51]: Good point will fix!

File: src/components/teamBoxes/TeamTwoBox.tsx (hunk 5 above)
Comment (Joshowens16) [reply src/components/teamBoxes/TeamTwoBox.tsx:49]: done!

File: src/components/teamBoxes/TeamTwoBox.tsx (hunk 9 above)
Comment (Joshowens16) [reply src/components/teamBoxes/TeamTwoBox.tsx:105]: taken out!

**Review by ocjarman — APPROVED** [review]
great! looking forward to using similar syntax for interfaces

**Pull request merged**